planetiler-core/src/main/java/com/onthegomap/planetiler/writer/TileArchive.java

```diff
@@ -0,0 +1,28 @@
+package com.onthegomap.planetiler.writer;
+
+import com.onthegomap.planetiler.config.PlanetilerConfig;
+import com.onthegomap.planetiler.util.LayerStats;
+
+/**
+ * A TileArchive is a portable on-disk stored tileset, e.g. MBTiles
+ */
+public interface TileArchive {
+
+  /**
+   * A high-throughput writer that accepts new tiles and queues up the writes to execute them in fewer large-batches.
+   */
+  public interface TileWriter extends AutoCloseable {
+    void write(TileEncodingResult encodingResult);
+
+    @Override
+    void close();
+
+    default void printStats() {}
+  }
+
+  void initialize(PlanetilerConfig config, TileArchiveMetadata metadata, LayerStats layerStats);
+
+  TileWriter newTileWriter();
+
+  void finalize(PlanetilerConfig config);
```

Comment:
can you add javadoc explaining what these do for callers and implementers?

Reply:
Maybe we should drop the `AutoCloseable` part of this interface? any cleanup that happens in `close` could be redundant with what's done in `finalize` though it's the caller's job to `finalize` - what vacuums the SQLite db or persists the PMTiles archive

Reply:
I think we want `AutoCloseable` (or `Closeable` since it throws an IOException), it's fine now since we have to do this to use it:

```java
    try (Mbtiles archive = Mbtiles.newWriteToFileDatabase(output, config.compactDb())) {
      TileArchiveWriter.writeOutput(featureGroup, archive, () -> FileUtils.fileSize(output), tileArchiveMetadata,
        config, stats);
    } catch (IOException e) {
      throw new IllegalStateException("Unable to write to " + output, e);
    }
```

but when there's a pmtiles version, we'll want to use try-with-resources to ensure that it closes, but we'll also need to be working with a generic reference to a `TileArchive`:

```java
    try (TileArchive archive = createTileArchive(output, config.compactDb())) {
      TileArchiveWriter.writeOutput(featureGroup, archive, () -> FileUtils.fileSize(output), tileArchiveMetadata,
        config, stats);
    } catch (IOException e) {
      throw new IllegalStateException("Unable to write to " + output, e);
    }
```

which needs TileArchive to implement closeable/autocloseable to compile